### PR TITLE
Feature/#46 backtracking

### LINKED ISSUE #46: Support backtracking
Provide the ability, to show previous game states.

### PR DESCRIPTION
**Pull request by DorianRec**
Fixes #46

### CODE REVIEW

**Review by jotoh98 — DISMISSED** [review]
tested on RandomAngleHintHider with NaiveAngleSearcher and GameManager - clicking next throws IllegalStateException from GameHistory.java:49

**Comment by DorianRec** [comment]
> tested on RandomAngleHintHider with NaiveAngleSearcher and GameManager - clicking next throws IllegalStateException from GameHistory.java:49

Fixed this now

File: src/main/java/com/treasure/hunt/game/GameEngine.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            return new Move(null, lastMovement, treasurePos);
+        } else {
+            lastHint = hider.move(lastMovement);
+        }
+        assert (lastHint != null);
+
+        if (!checkConsistency()) {
+            throw new IllegalStateException("Game is no longer consistent!");
+        }
+        System.out.println("" +
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/game/GameEngine.java]: use logger

File: src/main/java/com/treasure/hunt/game/GameEngine.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+     */
+    public Move init() {
+        searcherPos = JTSUtils.createPoint(0, 0);
+        searcher.init(searcherPos);
+
+        treasurePos = hider.getTreasureLocation();
+        assert (treasurePos != null);
+
+        // Check, whether treasure spawns in range of searcher
+        List<GeometryItem<Point>> act = new ArrayList<>();
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/game/GameEngine.java]: act is not a well chosen name

File: src/main/java/com/treasure/hunt/view/main/MainFrameController.java [hunk 3]
```diff
@@ -27,13 +30,19 @@ public void start() {
...
         CanvasView canvasView = new CanvasView();
-        GameManager gameManagerInstance = gameManager
-                .getDeclaredConstructor(Searcher.class, Hider.class, List.class)
-                .newInstance(newSearcher, newHider, Collections.singletonList(canvasView));
-        new CanvasController(canvasView, gameManagerInstance);
+        list.add(canvasView);
+        for (int i = 0; i < 7; i++) {
+            list.add(new DummyView());
+        }
+        /* TODO:
```
Comment (1-alex98) [inline src/main/java/com/treasure/hunt/view/main/MainFrameController.java]: well then do it

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

**Review by 1-alex98 — APPROVED** [review]

**Pull request merged**